gpiozero/boards.py

```diff
@@ -1391,15 +1417,27 @@ class PhaseEnableRobot(SourceMixin, CompositeDevice):
         A tuple of two GPIO pins representing the phase and enable inputs
         of the right motor's controller.
 
+    :param bool pwm:
+        If ``True`` (the default), construct :class:`PWMOutputDevice`
+        instances for the motor controller's enable pins, allowing both
+        direction and variable speed control. If ``False``, construct
+        :class:`DigitalOutputDevice` instances, allowing only direction
+        control.
+
     :param Factory pin_factory:
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
     """
 
-    def __init__(self, left=None, right=None, pin_factory=None):
+    def __init__(self, left=None, right=None, pwm=True, pin_factory=None, *args):
```

Comment:
Where the hell did `*args` and this comment come from?? I don't remember writing that

@lurch can you think of any reason this couldn't be taken out?

Reply:
> I don't remember writing that

:rofl: 
It _looks_ like you added that "args hack" so that `r = Robot(2, 3, 4, 5, pwm=True)` would give you a "friendly" error message of "raise GPIOPinMissing('left and right motor pins must be given as tuples')" rather than the (unfriendly?) error message of "TypeError: Robot() takes 5 positional arguments but 6 were given" ? :shrug: That's just a guess though.

But I dunno precisely how much hand-holding GpioZero should do to protect the user from making coding mistakes, so I've no objection to that being removed :+1:

Reply:
I don't understand how it would give you a friendly error message 😆

Reply:
It's not documented! Guess it's safe to be removed

Reply:
> I don't understand how it would give you a friendly error message

The way I understand it (which I should emphasise I'm still guessing about) is that with a class-prototype of `def __init__(self, left=None, right=None, pwm=True, pin_factory=None):` if you tried to do `r = Robot(2, 3, 4, 5, pwm=True)`, then python would assign left=2, right=3, pwm=True, pin_factory=4 and then get confused about the left-over 5 parameter, and so emit a TypeError. But with the class-prototype of `def __init__(self, left=None, right=None, pwm=True, pin_factory=None, *args):` I think python would then assign left=2, right=3, pwm=True, pin_factory=4, args=[5] and allow the code to continue on to where you check for `if type(left) is not tuple` ?
But I should qualify that all with a massive \*shrug\* ;)